components/ArticleList.jsx

```diff
@@ -19,8 +20,93 @@ const ArticleList = ({ topic_name }) => {
     return <h2>Loading...</h2>;
   }
 
+  function capFirstLetter(string) {
+    return string.charAt(0).toUpperCase() + string.slice(1);
+  }
+
+  const handleChange = (event) => {
+    const newInput = event.target.value;
+    setInput(newInput);
+    sortArticles(newInput);
+  };
+
+  const sortArticles = (input) => {
+    let sortedArticles = [...articles];
+
+    if (input === "date-asc") {
+      sortedArticles.sort((a, b) => {
+        const dateA = new Date(a.created_at);
+        const dateB = new Date(b.created_at);
+        return dateB - dateA;
+      });
+    }
+
+    if (input === "date-desc") {
+      sortedArticles.sort((a, b) => {
+        const dateA = new Date(a.created_at);
+        const dateB = new Date(b.created_at);
+        return dateA - dateB;
+      });
+    }
+
+    if (input === "comment-count-high") {
+      sortedArticles.sort((a, b) => b.comment_count - a.comment_count);
+    }
+
+    if (input === "comment-count-low") {
+      sortedArticles.sort((a, b) => a.comment_count - b.comment_count);
+    }
+
+    if (input === "votes-high") {
+      sortedArticles.sort((a, b) => b.votes - a.votes);
+    }
+
+    if (input === "votes-low") {
+      sortedArticles.sort((a, b) => a.votes - b.votes);
+    }
+
+    setArticles(sortedArticles);
+  };
```

Comment:
We built our back-end in order to be able to sort_by different categories, so we should take advantage of this instead of trying to sort it over here on the frontend.